src/SpasRequest.php

```diff
@@ -24,6 +24,12 @@ class SpasRequest implements ParsedRequest
     private $name = '';
 
     /**
+     * Name of the resource group, for hook purposes
+     * @var string
```

Comment:
The more I think about it, the greater the idea is, @cdamian 👍 

Let's `composer require hmaus/reynaldo` to get the interface `ApiResourceGroup` and add the resource group object rather than just the name. So your getter and setter would return a type of `ApiResourceGroup` and expect it for storage, instead of `string`.

Having the group on every request, would enable much more complex inspections later on, if required.
In hooks, for example, you could call `$request->getResourceGroup()->getTitle()` and even `$request->getResourceGroup()->getResources()` to build conditions around which other requests were in the same group.